x-pack/test/apm_api_integration/trial/tests/services/transaction_groups_charts.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { expectSnapshot } from '../../../common/match_snapshot';
+import { PromiseReturnType } from '../../../../../plugins/apm/typings/common';
+import { FtrProviderContext } from '../../../common/ftr_provider_context';
+import archives_metadata from '../../../common/archives_metadata';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+
+  const archiveName = 'apm_8.0.0';
+
+  const range = archives_metadata[archiveName];
+
+  // url parameters
+  const start = encodeURIComponent(range.start);
+  const end = encodeURIComponent(range.end);
+  const transactionType = 'request';
+
+  describe('APM Transaction Overview', () => {
+    describe('when data is loaded', () => {
+      before(() => esArchiver.load(archiveName));
+      after(() => esArchiver.unload(archiveName));
+
+      describe('and fetching transaction groups charts with uiFilters', () => {
+        const serviceName = 'opbeans-java';
+        let response: PromiseReturnType<typeof supertest.get>;
+
+        describe('without uiFilters', () => {
+          before(async () => {
+            response = await supertest.get(
+              `/api/apm/services/${serviceName}/transaction_groups/charts?start=${start}&end=${end}&transactionType=${transactionType}&`
+            );
+          });
+          it('should return an error response', () => {
+            expect(response.status).to.eql(400);
+          });
+        });
+
+        describe('when no environment is selected', () => {
+          const uiFilters = encodeURIComponent(JSON.stringify({}));
+          before(async () => {
+            response = await supertest.get(
+              `/api/apm/services/${serviceName}/transaction_groups/charts?start=${start}&end=${end}&transactionType=${transactionType}&uiFilters=${uiFilters}`
+            );
+          });
+
+          it('should have a successful response', () => {
+            expect(response.status).to.eql(200);
+          });
+
+          it('should return the ML job id for anomalies with no defined environment', () => {
+            expect(response.body).to.have.property('anomalyTimeseries');
+            expect(response.body.anomalyTimeseries).to.have.property('jobId');
+            expectSnapshot(response.body.anomalyTimeseries.jobId).toMatchInline(
+              `"apm-environment_not_defined-7ed6-high_mean_transaction_duration"`
```

Comment:
I'm not sure if this is what we intended for this API: Currently, no selected environment is functionally equivalent to selecting the `Not defined` option. Thoughts?

Reply:
Not specifying environment should not return anomalies.

Reply:
OK so this test should fail, I'll go ahead and fix the API, then update the test

Reply:
The problem occurs [here](https://github.com/elastic/kibana/blob/f20415e35d61aafdcf6a56ead00bf71d2fe3292d/x-pack/plugins/apm/server/lib/transactions/charts/get_anomaly_data/index.ts#L48-L51):
![image](https://user-images.githubusercontent.com/209966/94198882-47303280-feb8-11ea-9352-6b42fb36b44b.png)

The following is the easiest fix:
```ts
  if (!uiFilters.environment) {
```

However, I think the better solution (and @dgieselaar mentioned this too elsewhere) would be to make `environment` required from an API perspective, and keep the implementation as-is.

@ogupte Would you mind looking into making `environment` required? If it's too much work we can drop it. But would set us up nicely for eventually dropping the "ALL" option.

Reply:
OK i updated the API to make `environment` a required param, at the same time i made `uiFilters` an optional param since `serviceName` + `environment` are already required.